tck/features/PatternPredicateAcceptance.feature

```diff
@@ -0,0 +1,248 @@
+#
+# Copyright 2016 "Neo Technology",
+# Network Engine for Objects in Lund AB (http://neotechnology.com)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+Feature: PatternPredicateAcceptanceTest
+
+  Background:
+    Given an empty graph
+
+  Scenario: Filter relationships with properties using pattern predicate
+    And having executed:
+      """
+      CREATE ({node: 1})-[:X {rel: 1}]->({node: 2})
+      CREATE ({node: 3})-[:X {rel: 2}]->({node: 4})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE (n)-[{rel: 1}]->()
+      RETURN n.node AS id
+      """
+    Then the result should be:
+      | id |
+      | 1  |
+    And no side effects
+
+
+  Scenario: Filter using negated pattern predicate
+    And having executed:
+      """
+      CREATE ({node: 1})-[:X {rel: 1}]->({node: 2})
+      CREATE ({node: 3})-[:X {rel: 2}]->({node: 4})
+      """
+    When executing query:
+      """
+      MATCH (n) WHERE NOT (n)-[{rel: 1}]->()
+      RETURN n.node AS id
+      """
+    Then the result should be:
+      | id |
+      | 2  |
+      | 3  |
+      | 4  |
+    And no side effects
+
+  Scenario: Filter using a variable length relationship pattern predicate with properties
+    And having executed:
+      """
+      UNWIND [{node: 12, rel: 42}, {node: 324234, rel: 666}] AS props
+      CREATE (:START {p: props.node})-[:X {prop: props.rel}]->()-[:X {prop: props.rel}]->()
```

Comment:
Either we exclude `START` from the style checker, or we don't use it for anything except as a keyword. We should not have all-uppercase labels.